test/e2e/dnslookup_test.go

```diff
@@ -0,0 +1,299 @@
+		Addr:       testSrvAddr,
+		TCPPort:    dnsPort,
+		UDPPort:    dnsPort,
+		TCPHandler: tcpHandler,
+		UDPHandler: udpHandler,
+	}
+	t.Log("Starting test upstream DNS server")
+	go testServer.Run()
+
+	t.Logf("Starting host-resolver peer process in wsl [%v]", wslDistroName)
+	peerCmd := cmdExec(
+		tmpDir,
+		"wsl", "--user", "root",
+		"--distribution", wslDistroName,
+		"--exec", "./rancher-desktop-host-resolver", "vsock-peer")
+	err = peerCmd.Start()
+	require.NoError(t, err, "Starting host-resolver peer process faild")
+	defer func() {
+		_ = peerCmd.Process.Kill()
+	}()
+
+	t.Log("Starting host-resolver host process")
+	resolverExecPath := fmt.Sprintf("%v/rancher-desktop-host-resolver.exe", tmpDir)
+	hostCmd := cmdExec(
+		tmpDir,
+		resolverExecPath, "vsock-host",
+		"--upstream-servers", fmt.Sprintf("[%v]", testSrvAddr))
+	err = hostCmd.Start()
+	require.NoError(t, err, "Starting host-resolver host process faild")
+	defer func() {
+		_ = hostCmd.Process.Kill()
+	}()
+
+	t.Log("Building DNS hammer binary")
+	err = buildBinaries("../...", "linux", tmpDir)
+	require.NoError(t, err, "Failed building dnsHammer")
+
+	err = copyTestData("../testdata/test-300.csv", fmt.Sprintf("%s/test-300.csv", tmpDir))
+	require.NoError(t, err, "Failed copying test data file")
+
+	t.Log("Confirming host-resolver peer process is up")
+	peerCmdTimeout := time.Second * 10
+	confirm(func() bool { //nolint:errcheck // we don't care about the error
+		p, _ := os.FindProcess(peerCmd.Process.Pid)
+		return p.Pid == peerCmd.Process.Pid
+	}, tryInterval, peerCmdTimeout)
+
+	t.Logf("Running dns hammer test process in wsl [%v]", wslDistroName)
+	dnsSrvAddr := net.JoinHostPort(testSrvAddr, dnsPort)
+	runTestCmd := cmdExec(
+		tmpDir,
+		"wsl",
+		"--user", "root",
+		"--distribution", wslDistroName,
+		"--exec", "./test", "dnshammer",
+		"--server-address", dnsSrvAddr,
+		"--rr-type", "A=test-300.csv")
+	err = runTestCmd.Run()
+	require.NoError(t, err, "Running dns hammer against the peer process faild")
+	_ = runTestCmd.Process.Kill()
+}
+
+// TODO (Nino-K): maybe this can be replaced by CI
+func buildBinaries(path, goos, tmpDir string) error {
+	buildCmd := exec.Command("go", "build", "-o", tmpDir, path)
+	buildCmd.Env = os.Environ()
+	buildCmd.Env = append(buildCmd.Env, fmt.Sprintf("GOOS=%s", goos))
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+
+	return buildCmd.Run()
+}
+
+func cmdRunWithOutput(command string, args ...string) (string, error) {
+	var outBuf, errBuf strings.Builder
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = &outBuf
+	cmd.Stderr = &errBuf
+	err := cmd.Run()
+	if err != nil {
+		return errBuf.String(), err
+	}
+	return outBuf.String(), nil
+}
+
+func cmdExec(execDir, command string, args ...string) *exec.Cmd {
+	cmd := exec.Command(command, args...)
+	if execDir != "" {
+		cmd.Dir = execDir
+	}
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	return cmd
+}
+
+func copyTestData(src, dst string) error {
```

Comment:
```suggestion
func copyFile(src, dst string) error {
```
This doesn't rely on the file being test data.